Src/Base/AMReX_AsyncOut.cpp

```diff
@@ -38,17 +38,21 @@ void Initialize ()
     int nprocs = ParallelDescriptor::NProcs();
     s_noutfiles = std::min(s_noutfiles, nprocs);
 
+#ifdef AMREX_USE_MPI
     if (s_asyncout and s_noutfiles < nprocs)
     {
-#ifdef AMREX_MPI_THREAD_MULTIPLE
+        int provided = -1;
+        MPI_Query_thread(&provided);
+        if (provided < MPI_THREAD_MULTIPLE)
+            amrex::Abort("AsyncOut with " + std::to_string(s_noutfiles) + " and "
+                         +std::to_string(nprocs) + " processes requires "
+                         +"MPI_THREAD_MULTIPLE at runtime");
+
```

Comment:
I would add the provided thread safety level to the output message.

For example, Cori has an environment variable that must be set to allow THREAD_MULTIPLE support. Seeing if the support is 1 or 3 (SINGLE or SERIALIZED) could help diagnose that quickly.

Reply:
yes, will for that have to move your lambda to a more general location. I just implemented the same lookup in WarpX xD